subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/ui/SubscriptionWebViewViewModel.kt

```diff
@@ -290,13 +290,36 @@ class SubscriptionWebViewViewModel @Inject constructor(
         )
     }
 
-    private fun createOptionsJson(offer: SubscriptionOfferDetails, recurrence: String): OptionsJson {
+    private fun createOptionsJson(offer: SubscriptionOffer, recurrence: String): OptionsJson {
+        val offerDisplayPrice: String = offer.offerId?.let {
+            offer.pricingPhases.getOrNull(1)?.formattedPrice ?: offer.pricingPhases.first().formattedPrice
+        } ?: offer.pricingPhases.first().formattedPrice
+
         return OptionsJson(
             id = offer.planId,
             cost = CostJson(displayPrice = offer.pricingPhases.first().formattedPrice, recurrence = recurrence),
+            offer = getOfferJson(offer),
         )
     }
 
+    private fun getOfferJson(offer: SubscriptionOffer): OfferJson? {
+        return offer.offerId?.let {
+            val offerType = if (offer.pricingPhases.first().formattedPrice == "Free") {
```

Comment:
Isn't `formattedPrice` localized? Will comparison with `"Free"` work in other languages?

Reply:
Good question! 
It's a free trial type of offer, so I assumed that to identify them, `formattedPrice` would always come as 'Free' instead of a currency value. However, I could be wrong and it could be localised.
I can check if `priceAmountMicros` is 0 instead. I'll add that parameter to the model

Reply:
Synced offline and checking `offerId` is a free trial deal instead.